lib/viewmodels/new_comment_view_model.dart

```diff
@@ -0,0 +1,94 @@
+import "dart:async";
+
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/comment/comment_data.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/ui/new_comment_state.dart";
+import "package:proxima/services/database/comment_repository_service.dart";
+import "package:proxima/viewmodels/login_view_model.dart";
+
+/// The view model for adding a new comment to a post whose
+/// post id [PostIdFirestore] is provided as an argument.
+class NewCommentViewModel
+    extends AutoDisposeFamilyAsyncNotifier<NewCommentState, PostIdFirestore> {
```

Comment:
I don't think we want this notifier to be `AutoDispose` as if you make a small mistake and accidentally press "back" while typing a comment, you won't be able to get it back.

I think a simple solution would be that we keep the user's text on the comment field.
Otherwise, we could also have a small pop up asking for confirmation to leave the post page when the comment field is not empty.
I am more in favor of the first option, tell me what you think.

Reply:
You are totally right, thank you for pointing it out :+1: 

I chose to go with the first option which I find to be the less intrusive.
To do so, I removed the `AutoDispose` from the notifier and I moved the text field controller to be kept inside the viewmodel like that it is not destroyed when navigating back.

I implement this in 315d1e665f6bbf3f161dc814c63f37e57aab9abe.